app/distros/octane.js

```diff
@@ -0,0 +1,91 @@
+var request = require('request');
+
+function init() {
+  var module = {
+    id: 'octane',
+    profile: 'lightning'
+  };
+
+  module.option = {
+    name: 'Octane',
+    value: module.id
+  };
+
+  module.versions = [
+    {name: 'Octane 8.x', value: '8.x'}
+  ];
+
+  module.versionDefault = '8.x';
+
+  module.description = 'This project is built on [' + module.option.name + '] distribution.';
+
+  module.releaseVersion = function(majorVersion, done, cb) {
+    cb(null, '8.x', done);
+  };
+
+  module.drushMakeFile = function(yo, options, done) {
+    done();
+  };
+
+  // Merge keys from jsonB into jsonA.
+  // Existing values of jsonA are overwritten if they appear in jsonB.
+  // If recurse is true, call recursively for one level of subkeys.
+  function mergeJson(jsonA, jsonB, recurse) {
+    jsonA = (typeof jsonA !== 'undefined') ?  jsonA : {};
+    jsonB = (typeof jsonB !== 'undefined') ?  jsonB : {};
+    recurse = (typeof recurse !== 'undefined') ?  recurse : false;
+
+    var result = {};
+    for (var key in jsonA) {
+      result[key] = jsonA[key];
+    }
+    for (var key in jsonB) {
+      result[key] = jsonB[key];
+      if (recurse && (typeof jsonA[key] == 'object') && (typeof jsonB[key] == 'object')) {
+        // Only one level of recursion.
+        result[key] = mergeJson(jsonA[key], jsonB[key], false);
+      }
+    }
+    return result;
+  }
+
+  module.modifyComposer = function(yo, options, composer, composerExists, done) {
```

Comment:
Do you mean "Add comments about the philosophy here"?
The current code *does* merge them.  Anything *added* to the local composer.json is maintained.  However, for Octane, the drupal.org repo is the "source of record" for "require" and "patches" because this is how we want it to work so people can easily update their dependencies as Octane is updated.